app/controllers/reviewables_controller.rb

```diff
@@ -0,0 +1,180 @@
+class ReviewablesController < ApplicationController
+  requires_login
+
+  PER_PAGE = 10
+
+  before_action :version_required, only: [:update, :perform]
+
+  def index
+    min_score = params[:min_score].nil? ? SiteSetting.min_score_default_visibility : params[:min_score].to_i
+    offset = params[:offset].to_i
+
+    if params[:type].present?
+      unless ['ReviewableUser', 'ReviewableQueuedPost', 'ReviewableFlaggedPost'].include?(params[:type])
+        raise Discourse::InvalidParameter.new(:type)
+      end
+    end
+
+    status = (params[:status] || 'pending').to_sym
+    raise Discourse::InvalidParameter.new(:status) unless Reviewable.statuses[status].present?
+
+    category_id = nil
+    category_id = params[:category_id].to_i if params[:category_id]
+
+    topic_id = nil
+    topic_id = params[:topic_id].to_i if params[:topic_id]
+
+    total_rows = Reviewable.list_for(
+      current_user,
+      status: status,
+      category_id: category_id,
+      topic_id: topic_id,
+      min_score: min_score,
+    ).count
+    reviewables = Reviewable.list_for(
+      current_user,
+      status: status,
+      category_id: category_id,
+      topic_id: topic_id,
+      limit: PER_PAGE,
+      offset: offset,
+      type: params[:type],
+      min_score: min_score
+    ).to_a
```

Comment:
I considered that, but also figured you then have to deal with the overhead of a merge. It might be worth it? It was a "borderline" thing for me.

Reply:
IMO it depends if both calls are highly connected and the common options are unlikely to be changed only in one place. It's better to avoid duplication when possible

Reply:
I made this change and it does seem a lot easier to follow, at the minor cost of two extra `merge` calls which should be fairly fast.